modules/core/src/v2/coins/bsv.ts

```diff
@@ -47,31 +48,13 @@ export class Bsv extends Bch {
     return common.Environments[this.bitgo.getEnv()].bsvExplorerBaseUrl + url;
   }
 
-  getAddressInfoFromExplorer(addressBase58: string): Bluebird<any> {
-    const self = this;
-    return co(function* getAddressInfoFromExplorer() {
-      // TODO BG-9989: Update this method with the correct API route and parsing once we have one
-      const addrInfo = yield request.get(self.recoveryBlockchainExplorerUrl(`/addr/${addressBase58}`)).result();
-
-      addrInfo.txCount = addrInfo.txApperances;
-      addrInfo.totalBalance = addrInfo.balanceSat;
-
-      return addrInfo;
-    }).call(this);
+  getAddressInfoFromExplorer(addressBase58: string, apiKey?: string): Bluebird<any> {
+    const explorer = new BlockchairApi(this.bitgo, apiKey, 'bitcoin-sv');
+    return Bluebird.resolve(explorer.getAccountInfo(addressBase58));
   }
```

Comment:
use `async getAddressInfoFromExplorer(...)` and either `return explorer.getAccountInfo()` or `return await explorer.getAccountInfo()`

Reply:
This is bad advice. `getAddressInfoFromExplorer()` and `getUnspentInfoFromExplorer` are both part of the AbstractUtxoCoin API, which is public.

These functions are typed as follows:
```
protected abstract getAddressInfoFromExplorer(address: string, apiKey?: string): Bluebird<AddressInfo>;
protected abstract getUnspentInfoFromExplorer(address: string, apiKey?: string): Bluebird<UnspentInfo[]>;
```

Changing these signatures, or overriding these functions with a non-compatible signature should not be done unless you are intending to make breaking API changes and will increment the major version number accordingly.

Converting coroutines to async functions is a breaking change, since callers may be relying on bluebird specific methods being available on the returned promises. (eg, consider a call like `await getAddressInfoFromExplorer('someAddress').map((address) => /* do something with address */)`)

Coroutines can still be removed and replaced with async functions for code which is not exposed by the public API.

Please remove the `async` keyword from these function signatures.

Reply:
ah I see, yes we should stay compatible then return bluebird promises

however I think we can return bluebird promises *without* using coroutines 


are those two not equivalent?


```
f: co(function *() { return y })
```

```
async f() { return Bluebird.resolve(y) }
```

http://bluebirdjs.com/docs/api/promise.resolve.html

Reply:
Unfortunately, they are not.
```
$ node
> const bb = require('bluebird')
undefined
> async function promTest() { return bb.resolve('resolved'); }
undefined
> const bbTest = bb.coroutine(function*() { return 'resolved'; });
undefined
> promTest().map
undefined
> bbTest().map
[Function]
```

Reply:
The best way I've found to interoperate between different promise types while keeping the return value a bluebird is to use the `try()` method to start the promise chain. (more info: http://cryto.net/~joepie91/blog/2016/05/11/what-is-promise-try-and-why-does-it-matter/)